File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {

        vector <ListNode*> nodes;
        
        ListNode *curr = head;

        while(curr) {
            nodes.push_back(curr);
            curr = curr->next;
        }

        int rem = nodes.size() - n;

        if(rem == 0) {
            return head->next;
        }

        nodes[rem-1]->next = nodes[rem]->next;

        return head;


        
    }
};

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        


        # lists = [[1,2,4],[1,3,5],[3,6]]
        # result = [1,1,2,3,3,4,5,6]

        nodes = []

        for list in lists:
            while list:
                nodes.append(list.val)
                list = list.next
        
        nodes.sort()


        result = ListNode(0)
        curr = result

        for node in nodes:
            curr.next = ListNode(node)
            curr = curr.next

        return result.next
        